machine-learning/ml/preprocessing/preprocessing.py

```diff
@@ -65,6 +65,13 @@ def retrieve_labelled_instances(dataset, refactoring: LowLevelRefactoring):
     assert non_refactored_instances.shape[1] == refactored_instances.shape[1], "number of columns differ from both datasets"
     merged_dataset = pd.concat([refactored_instances, non_refactored_instances])
 
+    # Remove the faulty process metrics (with -1 as value)
+    log("instances before dropping faulty process metrics: {}".format(len(merged_dataset.index)))
+    merged_dataset = merged_dataset.query('''authorOwnership != -1 & bugFixCount != -1 & linesAdded != -1 & linesDeleted != -1\
```

Comment:
Why did you choose a "and" instead of "or" to filter incorrect process metrics? Why would we not consider them faulty if any of the metrices are not set?

Reply:
I changed all or's to and's in order to only select instances with non-faulty metrics.

Reply:
Also, I moved this after the feature reduction, in order to only drop instances where we consider the faulty process and authorship metrics for training.